tritium-samples/src/main/java/com/palantir/tritium/samples/Sample.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.tritium.samples;
+
+import java.util.Objects;
+
+public final class Sample {
+    private final long value;
+    private final long timestamp;
+    private final String traceId;
+
+    private Sample(long value, long timestamp, String traceId) {
+        this.value = value;
+        this.timestamp = timestamp;
+        this.traceId = traceId;
+    }
+
+    public static Sample of(long value, long timestamp, String traceId) {
+        return new Sample(value, timestamp, traceId);
+    }
+
+    public long getValue() {
+        return value;
+    }
+
+    public long getTimestamp() {
+        return timestamp;
+    }
+
+    public String getTraceId() {
+        return traceId;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null || getClass() != obj.getClass()) {
+            return false;
+        }
+        Sample sample = (Sample) obj;
+        return value == sample.value && timestamp == sample.timestamp && Objects.equals(traceId, sample.traceId);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(value, timestamp, traceId);
+    }
```

Comment:
it doesn't look like we're currently putting `Sample` in any hash based collections, but might be worth using allocation free `hashCode()` implementation to avoid issues down the road.

```suggestion
    @Override
    public int hashCode() {
        int result = Long.hashCode(getValue());
        result = 31 * result + Long.hashCode(getTimestamp());
        result = 31 * result + getTraceId().hashCode();
        return result;
    }
```